commit 8c5ba2e7215f8dc87271fbd3fcabcff8c8285cf2
Author: Zigit Zo <byte@kei.network>
Date:   Fri Apr 24 22:37:58 2026 +0800

    dbapi: vartree: prefix, only

diff --git a/lib/portage/dbapi/vartree.py b/lib/portage/dbapi/vartree.py
index 87d4f3453..a58d3f103 100644
--- a/lib/portage/dbapi/vartree.py
+++ b/lib/portage/dbapi/vartree.py
@@ -1805,6 +1805,7 @@ class dblink:
         errno.ENOTDIR,
         errno.EISDIR,
         errno.EPERM,
+        errno.EACCES,
     )
 
     def __init__(
@@ -2860,11 +2861,6 @@ class dblink:
                 file_data = pkgfiles[objkey]
                 file_type = file_data[0]
 
-                # don't try to unmerge the prefix offset itself
-                if len(obj) <= len(eroot) or not obj.startswith(eroot):
-                    show_unmerge("---", unmerge_desc["!prefix"], file_type, obj)
-                    continue
-
                 statobj = None
                 try:
                     statobj = os.stat(obj)
@@ -3288,9 +3284,12 @@ class dblink:
                 self._merged_path(os.path.realpath(parent_name), parent_stat)
 
                 show_unmerge("<<<", "", "dir", obj)
-            except OSError as e:
+            except (OSError, PermissionError) as e:
                 if e.errno not in ignored_rmdir_errnos:
                     raise
+                if e.errno == errno.EACCES:
+                    # no recursive, no revisit[]:
+                    show_unmerge("---", unmerge_desc["!prefix"], "dir", obj)
                 if e.errno != errno.ENOENT:
                     show_unmerge("---", unmerge_desc["!empty"], "dir", obj)
                     revisit[obj] = inode_key
@@ -5436,7 +5435,7 @@ class dblink:
             destroot,
             outfile,
             secondhand,
-            self.settings["EPREFIX"].lstrip(os.sep),
+            "", # emerge everything from image, like we've no prefix :)
             cfgfiledict,
             mymtime,
         ):
@@ -5795,35 +5794,6 @@ class dblink:
                         if dflags != 0:
                             bsd_chflags.lchflags(mydest, 0)
 
-                    if not stat.S_ISLNK(mydmode) and not os.access(mydest, os.W_OK):
-                        pkgstuff = pkgsplit(self.pkg)
-                        writemsg(
-                            _("\n!!! Cannot write to '%s'.\n") % mydest, noiselevel=-1
-                        )
-                        writemsg(
-                            _(
-                                "!!! Please check permissions and directories for broken symlinks.\n"
-                            )
-                        )
-                        writemsg(
-                            _(
-                                "!!! You may start the merge process again by using ebuild:\n"
-                            )
-                        )
-                        writemsg(
-                            "!!! ebuild "
-                            + self.settings["PORTDIR"]
-                            + "/"
-                            + self.cat
-                            + "/"
-                            + pkgstuff[0]
-                            + "/"
-                            + self.pkg
-                            + ".ebuild merge\n"
-                        )
-                        writemsg(_("!!! And finish by running this: env-update\n\n"))
-                        return 1
-
                     if stat.S_ISDIR(mydmode) or (
                         stat.S_ISLNK(mydmode) and os.path.isdir(mydest)
                     ):
