commit 8469c9efc1c4708fc739e18bd30e0942bd55e9c0
Author: Zigit Zo <byte@kei.network>
Date:   Wed Apr 22 20:58:04 2026 +0800

    xonsh: remove blink

diff --git a/xonsh/pyghooks.py b/xonsh/pyghooks.py
index f2e52545..fd4f8be5 100644
--- a/xonsh/pyghooks.py
+++ b/xonsh/pyghooks.py
@@ -142,8 +142,8 @@ def PYGMENTS_MODIFIERS():
         "FAINT": None,
         "ITALIC": "italic",
         "UNDERLINE": "underline",
-        "SLOWBLINK": "blink",
-        "FASTBLINK": "blink",
+        "SLOWBLINK": None,
+        "FASTBLINK": None,
         "INVERT": "reverse",
         "CONCEAL": "hidden",
         "STRIKETHROUGH": "strike",
@@ -151,7 +151,7 @@ def PYGMENTS_MODIFIERS():
         "FAINTOFF": None,
         "ITALICOFF": "noitalic",
         "UNDERLINEOFF": "nounderline",
-        "BLINKOFF": "noblink",
+        "BLINKOFF": None,
         "INVERTOFF": "noreverse",
         "REVEALOFF": None,
         "STRIKETHROUGHOFF": "nostrike",
